interface/src/app/gene/network/network.js

```diff
@@ -0,0 +1,272 @@
+        var minSvgSize = 600;  // Minimum size of svg.
+        var maxSvgSize = 1280; // Maximum size of svg.
+        var svgSize = genes.length * 10;
+        if (svgSize < minSvgSize) {
+          svgSize = minSvgSize;
+        } else if (svgSize > maxSvgSize) {
+          svgSize = maxSvgSize;
+        }
+
+        // Initialize tips on gene and edge.
+        var geneTip = d3.tip()
+            .attr('class', 'gene-tip')
+            .offset([-20, 0]);
+        var edgeTip = d3.tip()
+            .attr('class', 'edge-tip')
+            .offset([-20, 0]);
+
+        /**
+         * Compile gene information tips.
+         * @param {selected_gene_data} data;
+         * @return {string} The string in HTML format.
+         */
+        function getGeneInfo(data) {
+          var result = '<div id="title">' + data.label + '</div>';
+          result += '<br />Entrez ID: ' + data.entrezid;
+          if (data.systematic_name) {
+            result += '<br />Systematic name: ' + data.systematic_name;
+          }
+          if (data.standard_name) {
+            result += '<br />Standard name: ' + data.standard_name;
+          }
+          if (data.description) {
+            result += '<br />Description: ' + data.description;
+          }
+          if (data.aliases) {
+            result += '<br />aliases: ' + data.aliases;
+          }
+          return result;
+        }
+
+        /**
+         * Callback function to show edge tips when mouse is over an edge.
+         * @param {edge_data} data;
+         * @return {void}.
+         */
+        function showEdgeTip(data) {
+          var rawWeight = data.weight + rawMinWeight;
+          var str = 'Edge weight: ' + rawWeight.toFixed(3);
+          var heavyGenes = [data.gene1.id, data.gene2.id].join(',');
+          var target = d3.event.target;
+          NodeService.get(
+            {'heavy_genes': heavyGenes, 'limit': 0},
+            function success(response) {
+              var n = response.objects.length;
+              str += '<br />' + n + (n > 1 ? ' nodes are ' : ' node is ');
+              str += 'related to both genes' + (n > 0 ? ':' : '.');
+              for (var i = 0; i < n; ++i) {
+                str += '<br />* ' + response.objects[i].name + '</li>';
+              }
+
+              edgeTip.html(str);
+              edgeTip.show(data, target);
+            },
+            function error(err) {
+              $log.error('Failed to get node info for gene edge: ' + err);
+              str += 'Can\'t get node information, please try again later.';
+              edgeTip.html(str);
+              edgeTip.show(data, target);
+            }
+          );
+        }
+
+        /**
+         * Callback function to hide edge tips.
+         * @param {edge_data} data;
+         * @return {void}.
+         * ---------------------------------------------------------------------
+         * Since the edge tip is shown asynchronously, when edgeTip.hide() is
```

Comment:
For what it's worth: I needed a similar sort of workaround so the transitions would work right in the sample search results.